components/director/internal/domain/onetimetoken/service.go

```diff
@@ -236,13 +239,25 @@ func (s *service) getTokenFromAdapter(ctx context.Context, adapterURL string, ap
 	}
 
 	extTenant := tntCtx.ExternalID
+	var tnt *model.BusinessTenantMapping
 	if len(extTenant) == 0 {
-		extTenant, err = s.extTenantsSvc.GetExternalTenant(ctx, tntCtx.InternalID)
-		if err != nil {
-			return nil, errors.Wrapf(err, "while getting external tenant for internal tenant [%s]", tntCtx.InternalID)
+		if tnt, err = s.extTenantsSvc.GetTenantByID(ctx, tntCtx.InternalID); err != nil {
+			return nil, errors.Wrapf(err, "while getting tenant with internal ID %q", tntCtx.InternalID)
+		}
+	} else {
+		if tnt, err = s.extTenantsSvc.GetTenantByExternalID(ctx, tntCtx.ExternalID); err != nil {
+			return nil, errors.Wrapf(err, "while getting tenant with external ID %q", tntCtx.ExternalID)
```

Comment:
Is this if/else needed? We'll have the internal ID regardless and it'll be associated with an external tenant ID, so this branching of the code might not be needed and it'll make the mocking in the tests a bit easier, but in the end it's ok as is